pkg/virt-controller/services/sidecar.go

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package services
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/hooks"
+)
+
+type SidecarCreatorFunc func(*v1.VirtualMachineInstance, *v1.KubeVirtConfiguration) (hooks.HookSidecarList, error)
```

Comment:
nit: How about change the naming to have "Hook" prefix, it adds a little bit more  context about which sidecar we talking about:
`SidecarCreatorFunc` -> `HookSidecarCreatorFunc`
`WithSidecarCreator` -> `WithHookSidecarCreator`

Reply:
I think it creates a sidecar, not a hook. The hook is used to create sidecars.
I do not think it adds any more clarity if we add more names to it. Do you think it is colliding with some other sidecar thing?

Reply:
Currently there is no other refrance for sidecars in Kubevirt.
What I had in mind is that those sidecars related to hooks, so calling it hook-sidecar make sense to me. 
But its not a must, just a suggestion.